tests/tools/importexport/migration/test_migration.py

```diff
@@ -104,20 +106,24 @@ def test_migrate_recursively_specific_version(self):
         """Test the `version` argument of the `migrate_recursively` function."""
         filepath_archive = get_archive_file('export_v0.3_simple.aiida', **self.core_archive)
 
-        with Archive(filepath_archive) as archive:
+        reader_cls = get_reader(detect_archive_type(filepath_archive))
+        with reader_cls(filepath_archive) as archive:
+
+            metadata = deepcopy(archive._get_metadata())  # pylint: disable=protected-access
```

Comment:
Can't you just use
```suggestion
            metadata = deepcopy(archive.metadata)
```
?

Reply:
no because `.metadata` returns a dataclass (validated against the latest version), but here we want the "raw" file contents.
Really we shouldn't be using the reader class here, which intended to be used only with the latest vesion of the archive, rather a separate "migrator" class that can handle all versions.
But that is for another PR 😉